lib/csv.rb

```diff
@@ -1202,9 +1210,8 @@ def header_row?
   # The data source must be open for reading.
   #
   def shift
-    @parser_enumerator ||= parser.parse
     begin
-      @parser_enumerator.next
+      each.next
```

Comment:
This change causes performance regression with this benchmark: https://github.com/ruby/csv/blob/master/benchmark/shift.yaml

The following result is generated by changed YAML file to just show one case:

```console
% rake benchmark:shift:small
RUBYLIB= BUNDLER_ORIG_RUBYLIB= N_COLUMNS=10 /tmp/local/bin/ruby -v -S benchmark-driver /home/kou/work/ruby/csv/benchmark/shift.yaml
ruby 2.7.0dev (2019-02-02 trunk 66993) [x86_64-linux]
Warming up --------------------------------------
            unquoted    190.213 i/s -     200.000 times in 1.051452s (5.26ms/i)
Calculating -------------------------------------
                         master  shift-each 
            unquoted    187.657      76.664 i/s -     570.000 times in 3.037459s 7.435047s

Comparison:
                         unquoted
              master:       187.7 i/s 
          shift-each:        76.7 i/s - 2.45x  slower
```

We can resolve the regression by reverting this change.